ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelper.java

```diff
@@ -177,4 +268,11 @@ private void disconnect() throws XAException {
         }
     }
 
+    private void assertIfRecoveryScanWasStarted() throws XAException {
+        if (delegate == null) {
+            LOGGER.warn("XAResource.recover has to be called first with XAResource.TMSTARTRSCAN argument");
+            throw new XAException(XAException.XAER_PROTO);
```

Comment:
But how does that make it different? getXAResources always returns "this" and XARecoveryModule calls recover(XAResource.TMSTARTRSCAN) on it. Thus, in both cases connection will be always created by the same instance. And whichever method it is, it will be always called once. Same thing applies to closing.
In addition, this class is only used by XARecoveryModule and the only way that assertIfRecoveryScanWasStarted will fail is not because recover(XAResource.TMSTARTRSCAN) is not called, but because connection creation has failed (e.g. when Spring is shutting JMS before PeriodicRecovery finishes the scan).